### DISCUSSION TITLE
docker .net 6 image magic file

### DISCUSSION THREAD

**Discussion by devopg** [dotnet/dotnet-docker#3389] [Q&A] [answered]
hello i create default asp.mvc(core) .net 6 project in visual studio 2022 with checkbox "docker -linux". After docker build -> run image all work fine!

But i find in image magic file which take 129kb with name equal project name.

I havent this file in my project! I havent this file then i try without docker execute command: dotnet restore -> dotnet publish!

I try remove this file from container by hand and after restart all work fine without this file.

What is magic file ? How i can make docker images without this file? Intresting this **unnecessary** file **take 1/2 ALL MY aplication folder space!!!**

this file havent extension , this file is binaru (i think). list files:

![image](https://user-images.githubusercontent.com/61819835/148674466-29fa161f-2d81-4b8c-a305-44427fa40b05.png)

cat head: 

![image](https://user-images.githubusercontent.com/61819835/148674471-83076678-c649-40af-a5bf-889b5a83f14b.png)

**Comment by mthalman** [comment, marked as answer]
This is the platform-specific executable that gets output by default. You could use either `web1` or `dotnet web1.dll` to execute your application. This behavior is [documented](https://docs.microsoft.com/dotnet/core/deploying/#publish-framework-dependent):

> Publishing an app as framework-dependent produces a cross-platform binary as a dll file, and a platform-specific executable that targets your current platform.

You can prevent the output of this executable by set the `UseAppHost` property (include `/p:UseAppHost=false` when calling `dotnet publish`). I would argue that this should be set by default for these Dockerfiles since there is no intention to use the executable and the entrypoint is specifically set to use the DLL. So I've logged https://github.com/microsoft/DockerTools/issues/321 as a suggestion to use that property by default in the generated Dockerfiles.
